### PR TITLE
136 - Add GUI for password prompt

### PR DESCRIPTION
**Pull request by Thykof**
# What is added?
- [x] use fyne GUI library to prompt the password for signing a transaction
- [x] center the popup
- [x] focus the popup
- [x] bug: can't close thyra
- [x] build for macos with signature
- [ ] build for macos with notarize

Screenshot 2022-10-05 at 10 14 40

# Known issues
- in thyra app, step 1 is active (with ...) while the prompt is open (https://github.com/massalabs/thyra/issues/192)
- error handling when the user click on the Cancel button in the GUI is bad (https://github.com/massalabs/thyra/issues/193)

### CODE REVIEW

**Comment by Thykof** [comment]
@gregLibert I have squashed the commit you can merge as is if it's ok for you

**Review by Trottoir — APPROVED** [review]

File: int/api/server.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+	localAPI.ThyraWalletHandler = operations.ThyraWalletHandlerFunc(ThyraWalletHandler)
+	localAPI.ThyraWebsiteCreatorHandler = operations.ThyraWebsiteCreatorHandlerFunc(ThyraWebsiteCreatorHandler)
+
+	server.ConfigureAPI()
+
+	if err := server.Serve(); err != nil {
+		//nolint:gocritic
+		log.Fatalln(err)
+	}
```
Comment (gregLibert) [inline int/api/server.go:110]: To solve the interrupt signal is not stopping thyra:
```suggestion
	}
	(*app).Quit()
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -42,28 +48,74 @@ jobs:
           files: coverage.coverprofile
 
   build:
+    # if: ${{ github.ref == 'refs/heads/main' }}
```
Comment (thomas-senechal) [inline .github/workflows/CI.yml:51]: Is that commented line still useful ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by thomas-senechal — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -42,28 +48,74 @@ jobs:
...
     name: build and upload artifacts
+    env:
+      BUILD_CERTIFICATE_BASE64: ${{ secrets.BUILD_CERTIFICATE_BASE64 }}
+      P12_PASSWORD: ${{ secrets.P12_PASSWORD }}
+      BUILD_PROVISION_PROFILE_BASE64: ${{ secrets.BUILD_PROVISION_PROFILE_BASE64 }}
+      KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+      AC_USERNAME: ${{ secrets.AC_USERNAME }}
```
[the remark this answers is not in the document]
Comment (Thykof) [reply .github/workflows/CI.yml:58]: there are useful for signing and notarize, but for now we only sign. I suggest to keep like this and see in the next PR.

File: .github/workflows/CI.yml (hunk 2 above)
Comment (Thykof) [reply .github/workflows/CI.yml:51]: absolutely not, good catch

File: int/api/html/front/website.js [hunk 4]
```diff
@@ -8,10 +8,15 @@ let uploadable = false;
 getWallets();
 getWebsiteDeployerSC();
 initializeDefaultWallet();
-setupModal();
 
 const eventManager = new EventManager();
 
+async function onSubmitDeploy(txType = 'deployWebsiteAndUpload') {
+	setTxType(txType);
+	console.log("16");
```
[the remark this answers is not in the document]
Comment (Thykof) [reply int/api/html/front/website.js:16]: absolutely not, good catch

**Comment by thomas-senechal** [comment]
@Thykof As explained in [that comment](https://github.com/massalabs/thyra/pull/158#discussion_r992110248), I think there is an issue because most of the code you push is using tabulations instead of 4 spaces. Installating Prettier in your IDE should fix the problem.

**Pull request merged**